File: billing_library/src/main/java/com/robj/billinglibrary/Purchase.java
package com.robj.billinglibrary;

/**
 * Created by Rob J on 10/02/18.
 */

public class Purchase {

    private final com.android.billingclient.api.Purchase purchase;
    private final String orderId;
    private final String packageName;
    private final String sku;
    private final long purchaseTime;
    private final String purchaseToken;
    private final boolean autoRenewing;

    Purchase(com.android.billingclient.api.Purchase purchase) {
        this.purchase = purchase;
        this.orderId = purchase.getOrderId();
        this.packageName = purchase.getPackageName();
        this.sku = purchase.getSku();
        this.purchaseTime = purchase.getPurchaseTime();
        this.purchaseToken = purchase.getPurchaseToken();
        this.autoRenewing = purchase.isAutoRenewing();
    }

    public com.android.billingclient.api.Purchase getOriginalPurchase() {
        return purchase;
    }

    public String getOrderId() {
        return orderId;
    }

    public String getPackageName() {
        return packageName;
    }

    public String getSku() {
        return sku;
    }

    public long getPurchaseTime() {
        return purchaseTime;
    }

    public String getPurchaseToken() {
        return purchaseToken;
    }

    public boolean isAutoRenewing() {
        return autoRenewing;
    }
}

File: billing_library/src/main/java/com/robj/billinglibrary/models/Purchase.kt
package com.robj.billinglibrary.models

/**
 * Created by Rob J on 10/02/18.
 */

class Purchase internal constructor(val originalPurchase: com.android.billingclient.api.Purchase) {
    val orderId: String
    val packageName: String
    val sku: String
    val purchaseTime: Long
    val purchaseToken: String
    val isAutoRenewing: Boolean

    init {
        this.orderId = originalPurchase.orderId
        this.packageName = originalPurchase.packageName
        this.sku = originalPurchase.sku
        this.purchaseTime = originalPurchase.purchaseTime
        this.purchaseToken = originalPurchase.purchaseToken
        this.isAutoRenewing = originalPurchase.isAutoRenewing
    }
}
